network.go

```diff
@@ -1014,13 +1014,12 @@ func deleteNetNS(netNSPath string, mounted bool) error {
 	return nil
 }
 
-func createVirtualNetworkEndpoint(idx int, uniqueID string, ifName string) (*VirtualEndpoint, error) {
```

Comment:
You're modifying the behavior here, meaning we will end up with a new uniqueID for every endpoint. What you're doing is harmless but I'd like to understand where was the bug ?
Also, if we go with this solution, then I don't think we need to concatenate the unique ID with the actual ID.

Reply:
I wrote some description in the commit message.  Prior, the 'unique id' logic was being done before calling the create function.  Since this wasn't being used anywhere else, it makes more sense to do the exact same logic, only inside the function itself.

The bug that existed in the past was that the unique ID wasn't necessarily being created for each endpoint, as was intended (per discussion with @amshinde).  It was being called once, and then multiple virtual endpoints would potentially use this same "unique" ID.

Reply:
Yes, we were using one uniqueID per network namespace but the ID seems to me like it was unique since we were concatenating the uniqueID with the endpoint ID. I'll talk with @amshinde to understand the bug.

Reply:
Talking with @amshinde , it seems there was no bug, but a feeling there might be a bug :)
Ok anyway, I think we can keep this change but to go one step further, I think we should should concatenate with the endpoint ID, there is no point doing this anymore.

Reply:
Okay - - repushed.  Saw a build failure, though I don't see this as related to my changes.  Repushed in the hopes that it was a transient issue (I can build locally, and the same changes were fine on prior pushes) . Will try to check in on this later.  Thanks for the review, @sboeuf